src/command_modules/azure-cli-rdbms/tests/test_rdbms_commands.py

```diff
@@ -0,0 +1,379 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from datetime import datetime
+from dateutil.tz import tzutc
+
+from azure.cli.testsdk.base import execute
+from azure.cli.testsdk.exceptions import CliTestError
+from azure.cli.testsdk import (
+    JMESPathCheck,
+    NoneCheck,
+    ResourceGroupPreparer,
+    ScenarioTest)
+from azure.cli.testsdk.preparers import (
+    AbstractPreparer,
+    SingleValueReplacer)
+
+from azure.cli.testsdk.utilities import create_random_name
+
+
+# Constants
+SERVER_NAME_PREFIX = 'azuredbclitest'
+SERVER_NAME_MAX_LENGTH = 63
+
+
+class ServerPreparer(AbstractPreparer, SingleValueReplacer):
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, engine_type='mysql', engine_parameter_name='database_engine',
+                 name_prefix=SERVER_NAME_PREFIX, parameter_name='server', location='westeurope',
+                 admin_user='cloudsa', admin_password='SecretPassword123',
+                 resource_group_parameter_name='resource_group', skip_delete=True):
+        super(ServerPreparer, self).__init__(name_prefix, SERVER_NAME_MAX_LENGTH)
+        self.engine_type = engine_type
+        self.engine_parameter_name = engine_parameter_name
+        self.location = location
+        self.parameter_name = parameter_name
+        self.admin_user = admin_user
+        self.admin_password = admin_password
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.skip_delete = skip_delete
+
+    def create_resource(self, name, **kwargs):
+        group = self._get_resource_group(**kwargs)
+        template = 'az {} server create -l {} -g {} -n {} -u {} -p {}'
+        execute(template.format(self.engine_type,
+                                self.location,
+                                group, name,
+                                self.admin_user,
+                                self.admin_password))
+        return {self.parameter_name: name,
+                self.engine_parameter_name: self.engine_type}
+
+    def remove_resource(self, name, **kwargs):
+        if not self.skip_delete:
+            group = self._get_resource_group(**kwargs)
+            execute('az {} server delete -g {} -n {} --yes'.format(self.engine_type, group, name))
+
+    def _get_resource_group(self, **kwargs):
+        try:
+            return kwargs.get(self.resource_group_parameter_name)
+        except KeyError:
+            template = 'To create a rdbms server, resource group is required. Please add ' \
+                       'decorator @{} in front of this server preparer.'
+            raise CliTestError(template.format(ResourceGroupPreparer.__name__,
+                                               self.resource_group_parameter_name))
```

Comment:
I may have missed something obvious but `template` has one parameter but 2 params are passed to `template.format(...)`.